ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/AgentTool.java

```diff
@@ -135,4 +138,16 @@ public String getDefaultVersion() {
             return null;
         }
     }
+
+    private Map<String, String> extractFromChatParameters(Map<String, String> parameters) {
+        if (parameters.containsKey("input")) {
+            try {
+                Map<String, String> chatParameters = gson.fromJson(parameters.get("input"), Map.class);
+                parameters.putAll(chatParameters);
+            } finally {
```

Comment:
why `try...finally` used here without catch block?

Reply:
It is ok if we cannot parse result inside key `input`. The `input` is the key set by chat agent but can also be used by customer. So if we catch exception about parsing, I think we still do nothing. So I don't write catch block.

Reply:
How about change the method like:
```
    if (!parameters.containsKey("input")) {
        return;
    } 
    Map<String, String> chatParameters = gson.fromJson(parameters.get("input"), Map.class);
    parameters.putAll(chatParameters);
```

Reply:
Shouldn't we prompt this error to user?

Reply:
No it's ok. I'm a customer and directly use flow agent to correct my wrong json string. So it should be wrong.

Reply:
Can we add catch block and add logs in it? Make the log info level and do not log the stack trace, this might be helpful in some cases.

Reply:
Sure.

Reply:
Already do it.